src/realm/timestamp.hpp

```diff
@@ -75,6 +76,15 @@ class Timestamp {
         : m_is_null(true)
     {
     }
+    template <class T>
+    Timestamp(std::chrono::time_point<T> t)
+        : m_is_null(false)
+    {
+        auto time_since_epoc = t.time_since_epoch();
```

Comment:
We do not interact with the system clock. This just makes it possible to create a Timestamp from any std::chrono::time_point object.